CONTRIBUTING.md

```diff
@@ -49,15 +49,106 @@ If your pull request contains any of the below, it's less likely to be merged:
 **During pull request review**
 A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
 
-### Architecture
-The architecture of project is enforced with validation diagrams and will be run automatically at build. See the `eng/DependencyValidation` project to update or adjust this diagram. The following is currently enforced:
+## Architecture
+
+### Startup and hosting
+
+Upgrade Assistant is a .NET 5 console application that uses `System.CommandLine` APIs to [parse command line arguments](src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Program.cs) and then executes commands. Logging is done using `Microsoft.Extensions.Logging` APIs with a Serilog logging provider writing logs to the console.
```

Comment:
Some of this stuff seems like implementation details.... I'm worried about them falling out of sync over time

Reply:
Yea, I can see that. Even though some of this is implementation details, they're useful implementation details to record for someone new joining the project. I'll try to re-work it to point at source comments as much as possible, though. I agree that would be good.